### PR TITLE
arrays-arrays_with_methods-hard-q6

### PR DESCRIPTION
**Pull request by SamamthaH**
## PR Checklist
- [x]  Are helper methods really needed here?
    > Opt for hardcoding input/output into the input provider when possible
- [x]  Are the assertion fail messages good?
    - [x]  Logical (does it make sense?)
    - [x]  Readable (can a student understand what’s wrong?)
    - [x]  Not too feed-y (does it give away the answer?)
    - [x]  Grammatically sound (is the grammar and punctuation in the sentence proper?)
    - [x]  Specific (does the fail message relate to the question?)
- [x]  Is the code formatted well?
    > Run `CTRL` + `ALT` + `L` in IntelliJ
- [x]  EOF newlines
- [x]  Do the comments for Parsons follow the correct convention?
- [x]  Is the question being tested sufficiently?
    - [x]  Are there representative cases to test the program?
    - [x]  What about edge cases? (empty strings, negatives, infinity, off-by-one errors, etc)
        > If the question doesn’t specify what to do, mark it as a bad question
- [x]  Are the test method names good?
    > Test method names show up in a “What went well” section on the website, so “mathTest” will become “What went well: math test”, which doesn’t make much sense. A better name in this example would be “calculatesTaxCorrectly” so it shows up as “What went well: calculates tax correctly”
---
## Website
- [ ] Is the question text on the Gamification document up to date? 
    > Make sure to include the sample output if there is one for the question

### CODE REVIEW

File: src/arrays/arrays_with_methods/hard/q6/MainTest.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+public class MainTest {
+    // Java
+
+    public static Stream<Integer> randomReplaceInput(){
+        return Stream.of(3, 4, 5, 6, 7, 50);
+    }
+
+    @ParameterizedTest
+    @MethodSource("randomReplaceInput")
+    public void correctRandomReplaceMethod(int size) throws Throwable {
```
Comment (Hedgemon4) [inline src/arrays/arrays_with_methods/hard/q6/MainTest.java:24]: You don't need this return type, the custom assertion does this for you.

File: src/arrays/arrays_with_methods/hard/q6/MainTest.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        int repetitions = 0;
+        //get first round of numbers
+        for(int x = 0; x < size; x++){
+            if( x % 2 == 0 )
+                storeRandoms[count++][repetitions] = hold[x];
+        }
+        repetitions++;
+
+        //now rest of numbers
+        for(int x = 0; x < 999; x ++){
```
Comment (Hedgemon4) [inline src/arrays/arrays_with_methods/hard/q6/MainTest.java:51]: Also, we should separate these tests out and use an input provide like we normally do

**Review by Hedgemon4 — CHANGES_REQUESTED** [review]

File: src/arrays/arrays_with_methods/hard/q6/MainTest.java (hunk 1 above)
Comment (SamamthaH) [reply src/arrays/arrays_with_methods/hard/q6/MainTest.java:24]: we have a getMethodReturnType

File: src/arrays/arrays_with_methods/hard/q6/MainTest.java [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+public class MainTest {
+    // Java
+
+    public static Stream<Integer> randomReplaceInput() {
+        return Stream.of(3, 4, 5, 6, 7, 50);
+    }
+
+    @Test
+    public void randomReplaceMethodKeepsOddIndexNumber() throws Throwable {
```
Comment (Hedgemon4) [inline src/arrays/arrays_with_methods/hard/q6/MainTest.java:27]: This should probably be a parameterized test, as we should check this multiple times.

**Review by Hedgemon4 — CHANGES_REQUESTED** [review]

File: src/arrays/arrays_with_methods/hard/q6/MainTest.java [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+        int repetitions = 0;
+
+        //runs the program 1000 times and stores all the even index numbers generated. Each are in their own array
+        //because each index has its own range the variables can fall in (I'm effectively using ascending array to control that)
+        for (int x = 0; x < 1000; x++) {
+            for (int y = 0; y < size; y++) {
+                arr[y] = y + 1;
+            }
+            int count = 0;
+            int[] output = (int[]) m.callMethod();
```
Comment (Hedgemon4) [inline src/arrays/arrays_with_methods/hard/q6/MainTest.java:80]: If they don't return an array, will this not crash the program?

**Review by Hedgemon4 — CHANGES_REQUESTED** [review]

File: src/arrays/arrays_with_methods/hard/q6/MainTest.java (hunk 4 above)
Comment (SamamthaH) [reply src/arrays/arrays_with_methods/hard/q6/MainTest.java:80]: nope, the above
if (m.getMethodReturnType() != int[].class) {
            Assertions.fail("Your createRandomArray method does not return an array of integers.");
        }
is to make sure that if they don't return an int[] that the test fails before it crashes. Just tested it by making the method return void and the tests just returned a lot of "Your createRandomArray method does not return an array of integers." with no crashes

**Review by Hedgemon4 — APPROVED** [review]

**Pull request merged**